base/REPL.jl

```diff
@@ -74,6 +76,7 @@ function eval_user_input(ast::ANY, backend::REPLBackend)
             iserr, lasterr = true, (err, catch_backtrace())
         end
     end
+    ccall(:jl_sigatomic_end, Void, ())
```

Comment:
It isn't really needed in the sense that it should be unreachable. It is needed if we exit normally from the loop though so I included it for completeness (admittedly it won't help if one return from the loop).

It isn't dominated by a `begin`-`end` since the `leave` from the `try`-`catch` implicitly reset the defer_signal counter.